### PR TITLE
Refactor routes namespaces to move the helper functions out

### PR DESCRIPTION
**Pull request by drsnyder**
This is a follow-up to the feedback [here](https://github.com/Wikia/vignette/pull/75#discussion_r25633018).

This PR attempts to simplify our routing namespaces by moving the helper functions out into a separate namespace. This should reduce the clutter and make the main `http.routes` namespace more cohesive.

I also started an experiment with a very simple integration test to see if it works in travis. If it does we can add more to make sure that the API remains stable and ensure that the correct headers are in place. I think our test coverage is decent, but insufficient when making changes that affect the outermost part of the request stack like the handler as in the case with this PR.

/cc @nmonterroso

### CODE REVIEW

**Comment by drsnyder** [comment]
Duh, This would never work in travis. I'll make it so that you have to supply `:integration` explicitly to the command line to run the integration tests.

File: project.clj [hunk 1]
```diff
@@ -23,6 +23,7 @@
                                      [javax.servlet/servlet-api "2.5"]
                                      [org.clojure/tools.namespace "0.2.5"]
                                      [org.clojure/tools.trace "0.7.8"]
+                                     [clj-http "1.0.1"]
```
Comment (nmonterroso) [inline project.clj]: I've been trying so hard to keep these things in alphabetical order!

File: .midje.clj [hunk 2]
```diff
@@ -0,0 +1 @@
+(change-defaults :fact-filter (complement :thumbnail-integration))
```
Comment (nmonterroso) [inline .midje.clj]: wat is this

**Comment by nmonterroso** [comment]
LGTM

File: .midje.clj (hunk 2 above)
Comment (drsnyder) [inline .midje.clj]: It's midje's configuration for fact filtering. Is says that the default should not include that key. This is so those integration tests won't run in Travis.

File: project.clj (hunk 1 above)
Comment (drsnyder) [inline project.clj]: Ok, f34d0da.

**Pull request merged**